### ISSUE TITLE
Config object versus flat explicit arguments

### ISSUE THREAD

**Issue by thorwhalen** [i2mint/py2dash#5]
Consider the function:
```python
def dispatch_func_to_app(app, func, execute_on_change=False): ...
```

Compare the use of the execute_on_change argument to putting the argument as "VAR_KEYWORD" (e.g. **kwargs), or as the attribute of some config object.

dispatch_func_to_app is a general function that will do its job based on all kinds of specifications, of which execute_on_change is only one of them. This won't scale well, so we must think of the proper strategy.

# Explicit arguments

Using explicit arguments like `execute_on_change` above has definite advantages, of which:
* The user (developer) get's a more immediate understanding of the inputs (their types, defaults, meaning, etc.). 
* inspect.signature gives you more immediate information for static analysis (no need to resort to AST to infer the nature of inputs)

But if you have many variables, it becomes a bit much (e.g. many of the dash functions for html). 

_Note: Those who quote scripture might will want to quote the "explicit is better than implicit" commandment, but those who actually think will realize that many things depend on the context._

In our case, it's even worse than the "many variables", since the possibilities of "configuration" (specification of how to integrate the `func` to the `app`, in our example) are boundless. The `dispatch_func_to_app` will no doubt forward the work to other functions (that may have bounded and explicit arguments), but the function in our example will evolve, continuously. 

# So what do we do?

How we handle the configuration will need to be along the lines of a `**config_kwargs` or a `configs` object. 

Note that from there we can always offer explicit argument interfaces by using decorators that inject a specific signature for specific common use cases, but the main function implementing the configuration-based dispatching must be general. 

There's pros and cons in both, and many details about how configurations should be architectured. Let's discuss here. 

My tendency though, is to use a config object that contains it's parameters as parameters of the object. So our example above would look something like:

```python
class dflt_convention:
    execute_on_change=False

def dispatch_func_to_app(app, func, config=dflt_convention): ...
```

**Comment by SRHerzog** [comment]
I agree that it makes the most sense to have a single config object argument. This is the pattern I've seen most frequently in other libraries, it's familiar and convenient.
